SteamKit2/SteamKit2/Steam/CMClient.cs

```diff
@@ -83,6 +83,13 @@ public IPAddress LocalIP
         /// </value>
         public TimeSpan ConnectionTimeout { get; set; }
 
+        /// <summary>
+        /// Gets or sets whether or not this <see cref="CMClient"/> should imminently expect the server to close the connection.
+        /// If this is true when the connection is closed, the <see cref="SteamClient.DisconnectedCallback"/>'s <see cref="SteamClient.DisconnectedCallback.UserInitiated"/> property
+        /// will be set to <c>true</c>.
+        /// </summary>
+        public bool ExpectDisconnection { get; set; }
```

Comment:
Don't think this needs to be entirely public, consumers might be able to misuse it.